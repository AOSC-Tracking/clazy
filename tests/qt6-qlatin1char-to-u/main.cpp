#include <QtCore/QLatin1Char>
#include <QtCore/QChar>
#include <QtCore/QString>
#include <QtCore/QDir>

void receivingQChar(QChar s1) {}
void receivingQLatin1Char(QLatin1Char s1) {}

#define PREFIX '*'
void test()
{
    QChar c1 = QLatin1Char('*');
    QChar c11 = u'*'; // remove the check

    QString s = "aaa";
    bool b = s.startsWith(QLatin1Char('/'));

    s += QLatin1Char('.');

    QString appPrefix = s + QLatin1Char('\\');
    appPrefix = s + QLatin1Char('\'');

    if (s.at(1) == QLatin1Char('*'))
        b = true;

    QChar quotes[] = { QLatin1Char('"'), QLatin1Char('"') };
    QChar c2 = QLatin1Char(true ? '*' : '/');

    int i = s.lastIndexOf(QLatin1Char('*'));

    const QString sc = "sc";
    QString s2 = QDir::cleanPath(sc + QLatin1Char('*'));

    s2.insert(1, QLatin1Char('*'));


    receivingQChar(QLatin1Char('/'));

    receivingQLatin1Char(QLatin1Char('/'));
    //error: no matching function for call to 'receivingQLatin1Char'
    //receivingQLatin1Char(u'/'); // remove check

    QLatin1Char toto = QLatin1Char('/');
    //error: no viable conversion from 'char16_t' to 'QLatin1Char'
    //QLatin1Char totoo = u'/'; // remove check

    QChar char_with_macro = QLatin1Char(PREFIX); // should not be fixed
}
