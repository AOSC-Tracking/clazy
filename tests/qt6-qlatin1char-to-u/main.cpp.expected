qt6-qlatin1char-to-u/main.cpp:12:16: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:16:27: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:18:10: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:20:29: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:21:21: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:23:20: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:26:24: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:26:42: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:26:24: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:26:42: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:27:16: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:29:27: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:32:39: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:34:18: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:37:20: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:39:26: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:43:24: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
qt6-qlatin1char-to-u/main.cpp:47:29: warning: QLatin1Char(char) ctor being called [-Wclazy-qt6-qlatin1char-to-u]
